cortex/connection.py

```diff
@@ -25,47 +25,55 @@
 
 
 class ConnectionClient(_Client):
+    """A client used to manage connections on a Fabric server.
+
+    :param _Client: _description_
+    :type _Client: _type_
+    :return: An instance of a ConnectionClient
     """
-    A client used to manage connections.
-    """
-    URIs = {'connections': 'projects/{projectId}/connections'}
-
-    def __init__(self, *args, **kwargs):
-        super().__init__(*args, **kwargs)
-
-    def save_connection(self, connection: object):
-        """
-        Posts the connection client information.
-        :param connection: Connection object
-        :return: status
-        """
-        uri = self.URIs['connections'].format(projectId=self._project())
+
+    URIs = {"connections": "projects/{projectId}/connections"}
+
+    def save_connection(self, connection: dict) -> dict:
+        """Saves the provided `connection` dictionary to the Fabric API server.
+
+        :param connection: A Python dictionary containing specification for one of the `supported Fabric connection types <https://cognitivescale.github.io/cortex-fabric/docs/reference-guides/connection-types>`_
```

Comment:
Need to add broken link detection soon,  perhaps a post process if we can't do the during git hooks ?

Reply:
We should use Sphinx based external link resolution for this.. Will need a follow on ticket for this